Exceptions

asiajanicka opened the pull request:
Exceptions & logger

Commit:
Added exceptions

Commit:
Added some logging events

Commit:
Code recfactor, Changed flow in Trip Booking main, Added logs

vboltrukanis commented on src/main/java/exceptions/NoActivityException.java:
```diff
@@ -0,0 +1,10 @@
+package exceptions;
+
+public class NoActivityException extends Exception {
```
make exceptions different from each other, so there is some difference except for name

asiajanicka replied:
fixed.

vboltrukanis commented on src/main/java/trip/Participant.java:
```diff
@@ -119,6 +133,11 @@ public void setActivities(List<Activity> activities) {
 
+    @Override
+    public String toString() {
+        return person.toString();
```
Thats boring, lets change this)

asiajanicka replied:
fixed

vboltrukanis commented on src/main/java/trip/Participant.java:
```diff
@@ -33,11 +37,17 @@ public void addTransportBooking(TransportBooking transport) {
-            hotelBookings = new ArrayList<>();
+    public void addHotelBooking(HotelBooking hotelBooking) {
+        if(hotelBooking.calculatePrice() != null) {
+            if (hotelBookings == null) {
```
hotelBooking.calculatePrice should fail if hotelBooking is null, so maybe you reverse the order of checks?

asiajanicka replied:
In first IF I check if calculated price is null (in case of  lenght of staying in hotel is 0 or negative I set price to null)
In second IF I check if list of participant's hotel bookingS is null and if yes then I create new list and add the booking.
However, I will initialize the hotelBookingS list in the constructor.

Commit:
refactored code based on review feedback

Commit:
added generics in IFindPlacementByType; added org.jjm package for maven structure

Commit:
Removed generics; added remaining feedback from review

Commit:
Added IFindByType generic interface

Commit:
Replace CoachSeat & PlaneSeat with Seat<T>, removed IFindByTypeInterface, added generic method findSeatByType in Utils class (used in Flight & CoachTravel), changed TripBooking for tests

Commit:
refactored code, removed commented text, corrected class diagram

Commit:
Added reading hotel and rooms info from init data file for Malaga destination

Commit:
Added writing statistics about number of created objects to file

Commit:
moved prices from Board class to BoardType enum; deleted Board class

Commit:
moved luggage prices from FlightBooking class to PlaneBaggage enum

Commit:
changed Place class into enum

Commit:
moved enums closer to packages where they are used

Commit:
used lambda expresion with generics in Utils and Hotel class; refactored streams in Hotel and Transport

Commit:
Added util method to create Person from reflection

vboltrukanis commented on src/main/java/org/jjm/exceptions/PlacementAlreadyBooked.java:
```diff
@@ -0,0 +1,11 @@
+
+public class PlacementAlreadyBooked extends Exception{
+
+    public PlacementAlreadyBooked(){
```
Make exception more user-friendly and different from each other.

asiajanicka replied:
fixed

vboltrukanis commented on src/main/java/org/jjm/transport/Seat.java:
```diff
@@ -68,4 +74,5 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(number);
     }
+
```
not needed

asiajanicka replied:
removed

vboltrukanis commented on src/main/java/org/jjm/transport/Seat.java:
```diff
@@ -18,11 +21,14 @@ public Seat(int number, BigDecimal price) {
     }
 
+    @Override
     public boolean book() {
```
why dont you add unbook method, in case ticket was returned and the place is free now again.

asiajanicka replied:
added

vboltrukanis reviewed (changes requested):
Update your exception, rest is good. Mostly ready to merge

Commit:
Changed throwing runtime exception to system.exit

Commit:
Added method book to interface IBook and its implementation in Room and Seat classes
Moved logger messages from main flow to class methods
Rearranged try-caches for rooms and transport search in main flow of program
Customized exceptions, moved exception logger messages to inside exception classes

asiajanicka committed:
Delete tempStatistics.txt

Commit:
Changed generic T to extend Enum in Transport and Seat class; minor refactor

Commit:
removed runtime exceptions, added reading file paths from properties file

Commit:
corrected code style

vboltrukanis commented on src/main/java/org/jjm/hotel/Hotel.java:
```diff
@@ -0,0 +1,111 @@
+    }
+
+    public Room bookRoom(int roomNumber) throws NoPlacementException, PlacementAlreadyBooked {
+        if (getRoom(roomNumber).book())
```
isBooked()

vboltrukanis commented on src/main/java/org/jjm/hotel/exceptions/InvalidTimePeriodException.java:
```diff
@@ -0,0 +1,30 @@
+        logger.error(String.format(type.getDescription(), endDate, startDate), this.getStackTrace());
+    }
+
+
```
1 empty line between methods

vboltrukanis commented on src/main/java/org/jjm/hotel/exceptions/InvalidTimePeriodException.java:
```diff
@@ -0,0 +1,30 @@
+        super(message, cause);
+        logger.error(message, cause);
+    }
+
```
this line not needed

vboltrukanis reviewed (approved):
Minor changes. Plz check your code style once again before merge.

asiajanicka committed:
Merge pull request #8 from asiajanicka/reflections

Added util method to create Person from reflection

asiajanicka committed:
Merge branch 'files' into enum_lambda

asiajanicka committed:
Merge pull request #7 from asiajanicka/enum_lambda

Enum lambda

asiajanicka committed:
Merge branch 'generics' into files

Commit:
Fixes for conflict resolutions

asiajanicka committed:
Merge pull request #6 from asiajanicka/files

Files

Commit:
Merge branch 'exceptions' into generics

# Conflicts:
#	src/main/java/org/jjm/TripBooking.java
#	src/main/java/org/jjm/bookings/HotelBooking.java
#	src/main/java/org/jjm/destination/Destination.java
#	src/main/java/org/jjm/hotel/Hotel.java
#	src/main/java/org/jjm/hotel/Room.java
#	src/main/java/org/jjm/transport/Transport.java

asiajanicka committed:
Merge pull request #5 from asiajanicka/generics

Generics

Commit:
fixed code style
